asteroid/data/medleydb_dataset.py

```diff
@@ -0,0 +1,243 @@
+import torch
+from torch.utils import data
+import json
+import os
+import numpy as np
+import soundfile as sf
+import random
+import torchaudio
+
+
+class MedleydbDataset(data.Dataset):
+    """MedleyDB: a dataset of annotated, royaltyfree multitrack recordings.
+
+    It provides a stereo mix and both dry and processed multitrack stems for 
+    each song in the dataset. The dataset covers a wide distribution of genres 
+    and primarily consists of full length songs with professional 
+    or nearprofessional audio quality.
+    
+    The dataset consists of 254 full lengths music tracks of
+    different genres. It provides a stereo mix and both dry and processed
+    multitrack stems for each song. 
+
+    This dataset contains a .yml object for each multitrack containining
+    metadata for each audio file in the multitrack. This dataloader utilises 
+    this metadata to select audio files and downmix them to create mixtures
+    of instruments belonging to a list of instrument tags with specified 
+    polyphony.
+
+
+    Args:
+        json_dir (str): Path containing .json file with list of audio files
+        n_src (int): Number of separate instrument classes (yet to be implemented)
+        n_poly (int): Number of instances of each class to be present in mixture.
+        segment (float, optional): Duration of segments in seconds,
+            defaults to ``None`` which loads the full-length audio tracks.
+        sample_rate (int, optional): Samplerate of files in dataset.
+        threshold (float): activity confidence threshold to exclude segments with
+            less than given fraction of time of activity in segment.
+
+    References:
+        "MedleyDB: A Multitrack Dataset for Annotation-Intensive MIR Research", 
+            R. Bittner et. al. ISMIR 2014
+    
+    """
+
+    dataset_name = "MedleyDB"
+
+    def __init__(self, json_dir, n_src=1, n_poly=2, sample_rate=44100, segment=5.0, threshold=0.1):
+        super(MedleydbDataset, self).__init__()
+        
+        self.json_dir = json_dir
+        self.sample_rate = sample_rate
+        self.n_poly = n_poly
+        self.threshold = threshold
+        self.seg_len = segment
+        self.n_src = n_src
+        # Load json files
+        sources_json = [
+            os.path.join(json_dir, source + ".json")
+            for source in [f"inst{n+1}" for n in range(n_src)]
+        ]
+        sources_conf = []
+        for src_json in sources_json:
+            with open(src_json, "r") as f:
+                sources_conf = np.array(json.load(f))
```

Comment:
`sources_conf` is being overwritten every iteration, is that intended?

Reply:
For n_src (to be replaced by n_inst) = 1 this works fine, as the loop will only run once. For more than one inst I'll have to modify both sources_conf and sources_infos to have an extra dimension (do list append instead of =). I haven't tried that yet.